### PR TITLE
Fix: getBoolean function for Bit mysql data type and column Type Mism…

### PR DESCRIPTION
**Pull request by ashudeep-sharma**
…atch for BIGINT data type

This Pull Request contains the following changes:

- Implement getColumnTypeName() function for ResultSetMetadata.

- Fixing the Mapping for JDBC Type and MySql type for BIGINT.

- Handling the boolean data type for Bit(1) in mysql.

### CODE REVIEW

**Comment by enisoc** [comment]
@erzel and @dumbunny are the new reviewers for Java.

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSetMetaData.java [hunk 1]
```diff
@@ -99,7 +100,102 @@ public int getColumnType(int column) throws SQLException {
...
     public String getColumnTypeName(int column) throws SQLException {
         Query.Field field = getField(column);
-        return field.getType().toString();
+
+        int vitessTypeValue = field.getTypeValue();
+        int jdbcType = MysqlDefs.vitesstoMySqlType.get(field.getType());
+
+        switch (vitessTypeValue) {
+            case Query.Type.BIT_VALUE:
+                return "BIT";
```
Comment (dumbunny) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSetMetaData.java]: BIT_VALUE and DECIMAL_VALUE should be separated by a vertical space.

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSet.java [hunk 2]
```diff
@@ -208,6 +208,13 @@ public boolean getBoolean(int columnIndex) throws SQLException {
             return false;
         }
 
+        //Mysql 5.0 and higher have a BIT Data Type, need to check for this as well.
```
Comment (dumbunny) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSet.java]: Comment should have a leading space.

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSetMetaData.java [hunk 3]
```diff
@@ -99,7 +100,102 @@ public int getColumnType(int column) throws SQLException {
...
+
+            case Query.Type.ENUM_VALUE:
+                return "ENUM";
+
+            case Query.Type.YEAR_VALUE:
+                return "YEAR";
+
+            case Query.Type.SET_VALUE:
+                return "SET";
+
```
Comment (dumbunny) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSetMetaData.java]: Remove second vertical space.

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSetMetaData.java (hunk 1 above)
Comment (ashudeep-sharma) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSetMetaData.java]: Done.

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSet.java (hunk 2 above)
Comment (ashudeep-sharma) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSet.java]: Done.

File: java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSetMetaData.java (hunk 3 above)
Comment (ashudeep-sharma) [inline java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessResultSetMetaData.java]: Done.

**Comment by ashudeep-sharma** [comment]
@dumbunny : I don't have merge permissions, It seems like there is some intermittent failure in integration/travis-cli . Please verify it once and merge it when you have time.

**Comment by dumbunny** [comment]
Merged.

**Pull request merged**